users/users.go

```diff
@@ -0,0 +1,69 @@
+package users
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+)
+
+var ErrNotFound = errors.New("User not found")
+
+// User describes a user.
+type User struct {
+	ID       uint   `json:"id"`
+	UserName string `json:"userName"`
+}
+
+// DefaultDataProvider implements the DataProvider interface.
+type DefaultDataProvider struct{}
+
+// FindUserByID finds user by its ID, returns error if the user does not exist
+// or something goes wrong retrieving the user.
+func (t *DefaultDataProvider) FindUserByID(id uint) (*User, error) {
+	users, err := retrieveUsers()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, user := range users {
+		if user.ID == id {
+			return &user, nil
+		}
+	}
+	return nil, ErrNotFound
+}
+
+// FindUserByUserName finds user by its user name, return error if the user does
+// not exist or something goes wrong retrieving the user.
+func (t *DefaultDataProvider) FindUserByUserName(userName string) (*User, error) {
+	users, err := retrieveUsers()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, user := range users {
+		if user.UserName == userName {
```

Comment:
Could you please describe what is your criteria to use _variable_,  _varName_ or _VarName_ I am noticing the threes styles in this line.

Reply:
Of course.

Variables: simple camelCase. In this snippet, `use`r is a `User`, `userName` is a user name.
Fields (structs): exported fields (visible from outside the package) start with an uppercase letter, otherwise it will be visible only within this package.

Reply:
Excellent, we can add this to README just to be clear about the naming convention.

Reply:
That would be a lot, that's part of the language and there's already a README for that: https://github.com/golang/go/wiki/CodeReviewComments.

We could add that link to the README though.

Reply:
I want to complain about this 😏 😂